### PR TITLE
User Friend - Create & Update

### PR DESCRIPTION
**Pull request by smcunning**
### Description
PR adds CRUD functionality on the users friends endpoints to create and update a friend. Refactored to use facade & uses the genius serializer solution from Phil & Kate (thanks!) 

### Closes issue(s)

#1

### Testing
Happy path and some sad path testing complete for requests and facades.

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [x] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [ ] I ran rubocop and fixed issues
- [x] I ran full test suite - all tests passing

### CODE REVIEW

File: spec/requests/api/v1/friends/create_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      user_id: user_1.id,
+      friend_email: friend_email
+    }
+
+    results = JSON.parse(response.body, symbolize_names: true)
+
+    expect(results[:message]).to eq('unsuccessful')
+  end
+
+  it 'does not create a friendship if the user tries to friend themselves' do
```
Comment (katemorris) [inline spec/requests/api/v1/friends/create_spec.rb:35]: <3

**Review by katemorris — APPROVED** [review]
This looks GREAT!!!

File: app/controllers/api/v1/users/friends_controller.rb [hunk 2]
```diff
@@ -6,7 +6,15 @@ module Users
...
+          render json: FriendsSerializer.new(user)
+        end
+
+        def update
+          render json: FriendsFacade.accept_friendship(params[:user_id], params[:friend_id])
+        end
+
+        def create
+          render json: FriendsFacade.create_friendship(params[:user_id], params[:friend_email])
         end
```
Comment (PhilipDeFraties) [inline app/controllers/api/v1/users/friends_controller.rb:18]: I didn't realize a facade was something you could utilize for exposing endpoints, cool!

**Review by PhilipDeFraties — APPROVED** [review]
It seems as if this pr contains work that maybe could have been split into two prs so that they each focus on a specific functionality, but if its all working its working. Nice work!

**Pull request merged**